packages/@aws-cdk/aws-ec2/lib/vpc-endpoint.ts

```diff
@@ -0,0 +1,444 @@
+  public static readonly CodePipeline = new VpcEndpointAwsService('codepipeline');
+  public static readonly Config = new VpcEndpointAwsService('config');
+  public static readonly DynamoDb = new VpcEndpointAwsService('dynamodb', VpcEndpointType.Gateway);
+  public static readonly Ec2 = new VpcEndpointAwsService('ec2');
+  public static readonly Ec2Messages = new VpcEndpointAwsService('ec2messages');
+  public static readonly Ecr = new VpcEndpointAwsService('ecr.api');
+  public static readonly EcrDocker = new VpcEndpointAwsService('ecr.dkr');
+  public static readonly Ecs = new VpcEndpointAwsService('ecs');
+  public static readonly EcsAgent = new VpcEndpointAwsService('ecs-agent');
+  public static readonly EcsTelemetry = new VpcEndpointAwsService('ecs-telemetry');
+  public static readonly ElasticInferenceRuntime = new VpcEndpointAwsService('elastic-inference.runtime');
+  public static readonly ElasticLoadBalancing = new VpcEndpointAwsService('elasticloadbalancing');
+  public static readonly CloudWatchEvents = new VpcEndpointAwsService('events');
+  public static readonly ApiGateway = new VpcEndpointAwsService('execute-api');
+  public static readonly CodeCommitGit = new VpcEndpointAwsService('git-codecommit');
+  public static readonly CodeCommitGitFips = new VpcEndpointAwsService('git-codecommit-fips');
+  public static readonly KinesisStreams = new VpcEndpointAwsService('kinesis-streams');
+  public static readonly Kms = new VpcEndpointAwsService('kms');
+  public static readonly CloudWatchLogs = new VpcEndpointAwsService('logs');
+  public static readonly CloudWatch = new VpcEndpointAwsService('monitoring');
+  public static readonly S3 = new VpcEndpointAwsService('s3', VpcEndpointType.Gateway);
+  public static readonly SageMakerApi = new VpcEndpointAwsService('sagemaker.api');
+  public static readonly SageMakerRuntime = new VpcEndpointAwsService('sagemaker.runtime');
+  public static readonly SageMakerRuntimeFips = new VpcEndpointAwsService('sagemaker.runtime-fips');
+  public static readonly SecretsManager = new VpcEndpointAwsService('secretsmanager');
+  public static readonly ServiceCatalog = new VpcEndpointAwsService('servicecatalog');
+  public static readonly Sns = new VpcEndpointAwsService('sns');
+  public static readonly Sqs = new VpcEndpointAwsService('sqs');
+  public static readonly Ssm = new VpcEndpointAwsService('ssm');
+  public static readonly SsmMessages = new VpcEndpointAwsService('ssmmessages');
+  public static readonly Sts = new VpcEndpointAwsService('sts');
+  public static readonly Transfer = new VpcEndpointAwsService('transfer.server');
+
+  /**
+   * The name of the service.
+   */
+  public readonly name: string;
+
+  /**
+   * The type of the service.
+   */
+  public readonly type: VpcEndpointType;
+
+  constructor(name: string, type?: VpcEndpointType, prefix?: string) {
+    this.name = `${prefix || 'com.amazonaws'}.${cdk.Aws.region}.${name}`;
```

Comment:
Do we need to use some URL suffix token that's partition-aware (it's @RomainMuller's turf...)

Reply:
This is `AWS::URLSuffix` but reversed... don't know how does this look like for China regions...

Reply:
@RomainMuller our resident expert on the topic, thoughts?

Reply:
It's probably going to be different yet again for noncommercial regions (one of those things that looks like a domain name but really isn't). I think this needs to be in RegionInfo.

Reply:
Well all the documentation says is that those are formatted as `com.amazonaws.<region>.<service>`... And I couldn't find a reference table with more concrete information. So at this point I'd be willing to assume those are always formatted like so until proven wrong...